speech/check_dataVals.m

```diff
@@ -157,32 +180,73 @@ function reload_dataVals(src,evt)
     shortTrials = [];
     longTrials = [];
     nanFTrials = [];
-    jumpF1Trials = [];
-    jumpF2Trials = [];
-    fishyF1Trials = [];
+    for s = 1:UserData.nSigs
+        jumpTrials.(UserData.sigs2plot{s}) = [];
+    end
+    fishyTrials = [];
     earlyTrials = [];
     lateTrials = [];
     goodTrials = [];
 
     %% put trials into error categories
     for i = 1:length(dataVals)
+        %create vector of maximum differences, only for good trials and
+        %trials that are not just nans
+        maxDiffs = zeros(1,UserData.nSigs);
+        bNaNVals = all(isnan(dataVals(i).(UserData.sigs2plot{1})));
+        if strcmp(UserData.sigs2plot{s},'f0') %for f0 exclude first 40 ms of signal
+            %find trial with >1 sample
+            for j = 1:length(UserData.dataVals)
+                if length(UserData.dataVals(j).pitch_taxis) > 1
+                    tstep = diff(UserData.dataVals(j).pitch_taxis(1:2)); % get time step for each sample
+                    break;
+                end
+            end
+            onset = round(.04/tstep);
+        else
+            onset = 2; %exclude first sample from formant tracks when checking for NaNs
+        end
+        if ~dataVals(i).bExcl && ~bNaNVals
+            for s = 1:UserData.nSigs
+                if strcmp(UserData.sigs2plot{s},'f0') %for f0 exclude first 40 ms of signal
+                    onset = round(.04/tstep);
+                    if length(dataVals(i).(UserData.sigs2plot{s}))>onset+1
+                        maxDiffs(s) = max(abs(diff(dataVals(i).(UserData.sigs2plot{s})(onset:end))));
+                    elseif length(dataVals(i).(UserData.sigs2plot{s})) == 1 %can't diff only one sample
+                        maxDiffs(s) = NaN;
+                    else
+                        maxDiffs(s) = max(abs(diff(dataVals(i).(UserData.sigs2plot{s}))));
+                    end
+                else % for formants use whole track
+                    if length(dataVals(i).(UserData.sigs2plot{s})) == 1 %can't diff only one sample
+                        maxDiffs(s) = NaN;
+                    else
+                        maxDiffs(s) = max(abs(diff(dataVals(i).(UserData.sigs2plot{s}))));
+                    end
+                end
+            end
```

Comment:
```suggestion
            for s = 1:UserData.nSigs
                sig2plot = UserData.sigs2plot{s};
                if strcmp(sig2plot,'f0') %for f0 exclude first 40 ms of signal
                    onset = round(.04/tstep);
                    if length(dataVals(i).(sig2plot))>onset+1
                        maxDiffs(s) = max(abs(diff(dataVals(i).(UserData.sigs2plot{s})(onset:end))));
                    elseif length(dataVals(i).(sig2plot)) == 1 %can't diff only one sample
                        maxDiffs(s) = NaN;
                    else
                        maxDiffs(s) = max(abs(diff(dataVals(i).(sig2plot))));
                    end
                else % for formants use whole track
                    if length(dataVals(i).(sig2plot)) == 1 %can't diff only one sample
                        maxDiffs(s) = NaN;
                    else
                        maxDiffs(s) = max(abs(diff(dataVals(i).(sig2plot))));
                    end
                end
            end
```

Reply:
Improved readability in 4fac6c7d6d899be485b53d9b7596376fc69d8117